src/runtime-rs/crates/hypervisor/src/qemu/mod.rs

```diff
@@ -173,11 +173,11 @@ impl Hypervisor for Qemu {
 
     async fn guest_memory_block_size(&self) -> u32 {
         let inner = self.inner.read().await;
-        inner.guest_memory_block_size_mb()
+        inner.guest_memory_block_size()
     }
 
     async fn resize_memory(&self, new_mem_mb: u32) -> Result<(u32, MemoryConfig)> {
-        let inner = self.inner.read().await;
+        let mut inner = self.inner.write().await;
```

Comment:
I'm getting this build failure with rust 1.75 :
```
error: variable does not need to be mutable
   --> crates/hypervisor/src/qemu/mod.rs:180:13
    |
180 |         let mut inner = self.inner.write().await;
    |             ----^^^^^
    |             |
    |             help: remove this `mut`
    |
    = note: `-D unused-mut` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(unused_mut)]`
```

Compiler is right AFAICT  :wink:
```
    pub(crate) fn resize_memory(&self, _new_mem_mb: u32) -> Result<(u32, MemoryConfig)> {
```

Reply:
Right, the signature of `QemuInner::resize_memory()` is changed in the third commit where its implementation is added... this hunk can go to that commit if need be.